lib/cocoapods-binary-cache/helper/podspec.rb

```diff
@@ -1,8 +1,16 @@
 module Pod
   class Specification
-    # TODO: this detect objc lib as empty source, eg. Realm
     def empty_source_files?
-      return subspecs.all?(&:empty_source_files?) unless subspecs.empty?
+
+      if !subspecs.empty?
+        subspecs_empty = subspecs.all?(&:empty_source_files?)
+
+        # return early if there are some files in subpec(s)
+        # but process the spec itself
+        if !subspecs_empty
+          return false
+        end
+      end
```

Comment:
Oh, I got the flaw now 👍 

Minor feedback regarding code style:

```suggestion
      unless subspecs.empty?
        # return early if there are some files in subpec(s) but process the spec itself
        return false unless subspecs.all?(&:empty_source_files?)
      end
```